lib/MatchingAlgorithm.cc

```diff
@@ -626,10 +607,11 @@ void CriticalRSM::ties_propose(FreeListType& free_list, VertexPtr a, PreferenceL
     auto aj_data = bookkeep_data[aj];
     auto y = aj_data.level;
     auto star = aj_data.star;
-    int preference = prefers(b, a, aj);
+    auto pref_list_b = b->get_preference_list();
+    auto preference = pref_list_b.prefers(a, aj);
 
     if (a_data.level == t) {
-      if (y < t || ((y == t || (y == t && star)) && preference == 2)) {
+      if (y < t || ((y == t || (y == t && star)) && preference == cBetter)) {
```

Comment:
Complex statements should be wrapped in round braces, like (y==t) and (preference == cBetter)